lms/djangoapps/grades/rest_api/v1/views.py

```diff
@@ -207,3 +217,307 @@ def _get_course(self, request, course_id):
+            empty 'results' field.
+        """
+        course_grading_status = []
+        course_keys = []
+        username_filter = None
+
+        form = CourseEnrollmentsApiListForm(self.request.query_params)
+        if not form.is_valid():
+            raise ValidationError(form.errors)
+        usernames = form.cleaned_data.get('username')
+
+        if self.request.query_params.get('course_id'):
+            try:
+                course_keys.append(get_course_key(request, course_id))
+            except InvalidKeyError:
+                raise self.api_error(  # lint-amnesty, pylint: disable=raise-missing-from
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    developer_message='The provided course key cannot be parsed.',
+                    error_code='invalid_course_key'
+                )
+        else:
+            course_keys = self._get_enrollment_course_keys()
+        if usernames:
+            username_filter = [Q(user__username__in=usernames)]
+        for course_key in course_keys:
+            users = self._paginate_users(course_key, course_enrollment_filter=username_filter)
+            with bulk_course_grade_context(course_key, users):
+                for user, course_grade, exc in CourseGradeFactory().iter(users, course_key=course_key):
+                    if not exc:
+                        course_grading_status.append(
+                            self._serialize_course_grading_status(user, course_key, course_grade)
+                        )
+        return self.get_paginated_response(course_grading_status)
+
+    def _get_enrollment_course_keys(self):
+        """ Returns all unique course_keys for enrollments.
+        """
+        unique_course_string_map = {}
+        course_ids = CourseEnrollment.objects.select_related('course').all().values_list('course_id', flat=True)
+        for course_id in course_ids:
+            unique_course_string_map[str(course_id)] = course_id
+        return list(unique_course_string_map.values())
+
+
+@can_disable_rate_limit
+class SubmissionHistoryView(GradeViewMixin, PaginatedAPIView, ApiKeyPermissionMixIn):
+    """
+    Submission history view.
+    """
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (permissions.IsAuthenticated,)
+    throttle_classes = (EnrollmentUserThrottle,)
+
+    def get(self, request, course_id=None, username=None):
+        """
+        Get submission history details.
+
+        **Usecases**:
+
+            Regular users can only retrieve their own submission history and users with GlobalStaff status
+            can retrieve everyone's submission history.
+
+        **Example Requests**:
+
+            GET /api/grades/v1/submission_history?course_id=course_id
+            GET /api/grades/v1/submission_history?course_id=course_id&user=username
```

Comment:
```suggestion
            GET /api/grades/v1/submission_history?course_id=course_id&username={username}
```

Also, please use a consistent format for {variables} in examples.